internal/errors/compressor_test.go

```diff
@@ -0,0 +1,221 @@
+package errors
+
+import (
+	"testing"
+)
+
+func TestErrInvalidCompressionLevel(t *testing.T) {
+	type args struct {
+		level int
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "returns error when compression leve is 100",
+			args: args{
+				level: 100,
+			},
+			want: want{
+				want: New("invalid compression level: 100"),
+			},
+		},
+		{
+			name: "returns error when compression leve is 0",
+			args: args{
+				level: 0,
```

Comment:
```suggestion
				level: math.Min(int),
```

Reply:
also, could you please define value as minimum number of int?

Reply:
I got it